tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs

```diff
@@ -1230,15 +1229,19 @@ public void TestFindAndRemoveWithFields()
         [Fact]
         public void TestFindAndRemoveWithMaxTime()
         {
-            if (_primary.Supports(FeatureId.MaxTime))
+            var server = GetServerFromDisposableClient();
```

Comment:
I didn't `dispose` this client because in this case, next test call would fail with the error: 

    System.ObjectDisposedException : Cannot access a disposed object.
Object name: 'MultiServerCluster'

It's related to the fact that a `disposable client` and `LegacyTestConfiguration.Server` use the same MongoClient static dictionary:
https://github.com/mongodb/mongo-csharp-driver/blob/7cd604ab4fee1c3a02fd56e7b3f99215055785f4/src/MongoDB.Driver.Legacy/MongoServer.cs#L64

Reply:
To use `using` we need to remove disposed server via `MongoServer.UnregisterServer(server)`, not sure that it's a better way than the currently implemented

Reply:
Do this instead:

    var server = LegacyTestConfiguration.GetServer(retryWrites: false);

Reply:
Done